tracee-ebpf/tracee/process_tree.go

```diff
@@ -0,0 +1,174 @@
+package tracee
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/aquasecurity/tracee/pkg/containers"
+)
+
+type ProcessCtx struct {
+	StartTime   int
+	ContainerID string
+	Pid         int
+	Tid         int
+	Ppid        int
+	HostPid     int
+	HostTid     int
+	HostPpid    int
+	Uid         int
+	MntId       int
+	PidId       int
+}
+
+type ProcessTree struct {
+	processTreeMap map[int]ProcessCtx
+}
+
+func (t *Tracee) SetProcessContext(ctx []byte) ProcessCtx {
+	var procCtx = ProcessCtx{}
+	procCtx.StartTime = int(binary.LittleEndian.Uint64(ctx[0:8]))
+	cgroupId := binary.LittleEndian.Uint64(ctx[8:16])
+	procCtx.ContainerID = t.containers.GetCgroupInfo(cgroupId).ContainerId
+	procCtx.Pid = int(binary.LittleEndian.Uint32(ctx[16:20]))
+	procCtx.Tid = int(binary.LittleEndian.Uint32(ctx[20:24]))
+	procCtx.Ppid = int(binary.LittleEndian.Uint32(ctx[24:28]))
+	procCtx.HostPid = int(binary.LittleEndian.Uint32(ctx[28:32]))
+	procCtx.HostTid = int(binary.LittleEndian.Uint32(ctx[32:36]))
+	procCtx.HostPpid = int(binary.LittleEndian.Uint32(ctx[36:40]))
+	procCtx.Uid = int(binary.LittleEndian.Uint32(ctx[40:44]))
+	procCtx.MntId = int(binary.LittleEndian.Uint32(ctx[44:48]))
+	procCtx.PidId = int(binary.LittleEndian.Uint32(ctx[48:52]))
+	return procCtx
+}
+
+//returns the file creation time
+//Note: this is not the real process start time due its based on the file creation time, but it is a close approximation
+func processStartTime(path string) (time.Time, error) {
```

Comment:
I think this should just be "getFileCreationTime", and just assign it to a variable named "processStartTime"